components/mediation/data-publishers/org.wso2.micro.integrator.prometheus.publisher/src/main/java/org/wso2/micro/integrator/prometheus/publisher/util/MetricFormatter.java

```diff
@@ -57,7 +54,7 @@ private String typeString(Type type) {
      * @param metricsList the Prometheus Metric from Prometheus Default Registry
      * @return the metric converted into String format
      */
-    public StringBuilder formatMetrics(Enumeration<MetricFamilySamples> metricsList) {
+    public static StringBuilder formatMetrics(Enumeration<MetricFamilySamples> metricsList) {
```

Comment:
Do not return the string builder. Return the string. StringBuilders should be kept as method level object. Else they tend to cause memory leaks

Reply:
Fixed